### PR TITLE
Input hydrator (tmp)

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: blog/src/Blog/Post/PostController.php [hunk 1]
```diff
@@ -37,7 +38,7 @@ public function index(CurrentRoute $currentRoute, PostRepository $postRepository
         return $this->viewRenderer->render('index', ['item' => $item, 'canEdit' => $canEdit, 'slug' => $slug]);
     }
 
-    public function add(Request $request, ValidatorInterface $validator): Response
+    public function add(Request $request, FormHydrator $formHydrator): Response
```
Comment (samdark) [inline blog/src/Blog/Post/PostController.php:41]: How about calling it InputHydrator? Or is it form-specific?

File: blog/src/Blog/Post/PostController.php [hunk 2]
```diff
@@ -48,15 +49,13 @@ public function add(Request $request, ValidatorInterface $validator): Response
...
-                    ->validate($form)
-                    ->isValid()) {
+            if ($formHydrator->populate($form, $parameters['body']) && $form->isValid()) {
                 $this->postService->savePost($this->userService->getUser(), new Post(), $form);
 
                 return $this->webService->getRedirectResponse('blog/index');
             }
 
-            $parameters['errors'] = $form->getFormErrors();
+            $parameters['errors'] = $form->getValidationResult()->getErrorMessagesIndexedByAttribute();
```
Comment (samdark) [inline blog/src/Blog/Post/PostController.php:58]: Looks more complicated.

File: blog/src/Blog/Post/PostController.php (hunk 1 above)
Comment (vjik) [reply blog/src/Blog/Post/PostController.php:41]: It's form specific hydrator — get name from `Form` object and populate by this name.

File: blog/src/Blog/Post/PostController.php (hunk 2 above)
Comment (vjik) [reply blog/src/Blog/Post/PostController.php:58]: There are several cases of errors usage:
- simple list of errors
- errors indexed by attribute
- errors indexed by path

All cases implemented in `Result` object. Should we duplicate this methods in form class ?

File: blog-api/src/Auth/AuthRequest.php [hunk 3]
```diff
@@ -17,25 +18,31 @@
  *      @OA\Property(example="Opal1144", property="password", format="string"),
  * )
  */
-final class AuthRequest extends RequestModel implements RulesProviderInterface
```
Comment (samdark) [inline blog-api/src/Auth/AuthRequest.php:20]: Do we still need request model package then in `composer.json`?

**Review by samdark — APPROVED** [review]

File: blog-api/src/Auth/AuthRequest.php (hunk 3 above)
Comment (vjik) [reply blog-api/src/Auth/AuthRequest.php:20]: Not need. Removed request model dependency.

**Comment by vjik** [comment]
Closed in favor #599

**Pull request closed without merging**